File: leetcode/204. Count Primes/204._Count_Primes_1.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 2:
            return 0
        isPrime = [True for i in xrange(n)]
        isPrime[0], isPrime[1] = False, False
        
        i = 2
        while i*i < n:
            if isPrime[i] == False:
                i += 1
                continue
            for j in xrange(i*i, n, i):
                isPrime[j] = False
            i += 1
        return isPrime.count(True)

File: leetcode/204. Count Primes/204._Count_Primes_1.java
public class Solution {
    public int countPrimes(int n) {
        if(n<1)
            return 0;
        boolean[] flag = new boolean[n+1];
        flag[0] = true;
        flag[1] = true;
        for(int i=2; i*i<n; i++) {
            if(flag[i])
                continue;
            int c = i*i;
            while(c < n) {
                flag[c] = true;
                c += i;
            }
        }
        int count = 0;
        for(int i=0; i<n; i++) {
            if(!flag[i])
                count++;
        }
        return count;
    }
}